src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1591,39 +1592,59 @@ public SolutionCompilationState RemoveAnalyzerConfigDocuments(ImmutableArray<Doc
 
         foreach (var documentIdsInProject in documentIdsByProjectId)
         {
-            var oldProjectState = this.SolutionState.GetProjectState(documentIdsInProject.Key);
+            newCompilationState = newCompilationState.RemoveDocumentsFromSingleProject<T>(documentIdsInProject.Key, [.. documentIdsInProject]);
+        }
 
-            if (oldProjectState == null)
-            {
-                throw new InvalidOperationException(string.Format(WorkspacesResources._0_is_not_part_of_the_workspace, documentIdsInProject.Key));
-            }
+        return newCompilationState;
+    }
 
-            using var _ = ArrayBuilder<T>.GetInstance(out var removedDocumentStates);
+    private SolutionCompilationState RemoveDocumentsFromSingleProject<T>(ProjectId projectId, ImmutableArray<DocumentId> documentIds)
```

Comment:
>ImmutableArray [](http://example.com/codeflow?start=94&length=14)

nice, it looks like you got rid of a ToLookup call that was being executed twice.